File: __tests__/model/hand/detector/hasFlush.ts
import hasFlush from "../../../../src/model/hand/detect/hasFlush";
import cards from "../../../../src/model/card/cards";

describe("detect flushes", () => {
  test("detect a flush", () => {
    expect(hasFlush(cards("D2", "D4", "DQ", "DJ", "DT"))).toBe(true);
  });

  test("detect a flush from seven cards", () => {
    expect(hasFlush(cards("D2", "D4", "DQ", "DJ", "DT", "H3", "H2"))).toBe(
      true
    );
  });

  test("detect a flush from seven cards of equal suit", () => {
    expect(hasFlush(cards("D2", "D4", "DQ", "DJ", "DT", "D3", "D5"))).toBe(
      true
    );
  });
});

File: src/rust/src/tests/has_flush.rs
use crate::card::Card;
use crate::hand::Hand;

#[test]
fn test_has_flush() {
    // Create a hand with a flush
    let cards = vec![
        Card::new(crate::card::Suit::Hearts, crate::card::Value::Ace),
        Card::new(crate::card::Suit::Hearts, crate::card::Value::King),
        Card::new(crate::card::Suit::Hearts, crate::card::Value::Queen),
        Card::new(crate::card::Suit::Hearts, crate::card::Value::Jack),
        Card::new(crate::card::Suit::Hearts, crate::card::Value::Ten),
        Card::new(crate::card::Suit::Spades, crate::card::Value::Nine),
        Card::new(crate::card::Suit::Spades, crate::card::Value::Eight),
    ];

    let hand = Hand::new(cards);
    assert!(hand.has_flush());

    // Create a hand without a flush
    let cards = vec![
        Card::new(crate::card::Suit::Hearts, crate::card::Value::Ace),
        Card::new(crate::card::Suit::Diamonds, crate::card::Value::King),
        Card::new(crate::card::Suit::Clubs, crate::card::Value::Queen),
        Card::new(crate::card::Suit::Spades, crate::card::Value::Jack),
        Card::new(crate::card::Suit::Hearts, crate::card::Value::Ten),
        Card::new(crate::card::Suit::Spades, crate::card::Value::Nine),
        Card::new(crate::card::Suit::Diamonds, crate::card::Value::Eight),
    ];

    let hand = Hand::new(cards);
    assert!(!hand.has_flush());
}
